### ISSUE TITLE
Huge performance degradation on different CPU

### ISSUE THREAD

**Issue by Dockyin** [tensorflow/tensorflow#45783]
I have a model with ResNet50 which is trained by Keras.
This model can be run on CPU "Intel(R) Xeon(R) Gold 6138T CPU @ 2.00GHz" in 1.2 second, but when predict on "E5 2620" it need 15s.
Is there any idea to solve the performance degradation? Thanks.
P.S. Both of the 2 cases is in virtual mechine.

**Comment by amahendrakar** [comment]
@Dockyin,
Could you please provide the TensorFlow version you are using and a minimal code snippet to reproduce the issue?

Please go through [this](https://www.tensorflow.org/guide/profiler) guide to optimize TensorFlow performance and also take a look at the guide to [debug performance bottlenecks](https://www.tensorflow.org/tensorboard/tensorboard_profiling_keras#debug_performance_bottlenecks) and check if it helps. Thanks!

**Comment by wei-v-wang** [comment]
E5 2620 are broadwell architecture systems while Intel Xeon Gold 6138 is Skylake which is newer generation than Broadwell (added AVX512 instruction set support). So it is expected that 6138 would be better than E5 2620. 

But, are you using the "pip install tensorflow-mkl" wheel and with that, set the correct OpenMP threads? 

Adding @TensorFlow-MKL  for further trouble-shooting.

**Comment by Dockyin** [comment]
> @Dockyin,
> Could you please provide the TensorFlow version you are using and a minimal code snippet to reproduce the issue?
> 
> Please go through [this](https://www.tensorflow.org/guide/profiler) guide to optimize TensorFlow performance and also take a look at the guide to [debug performance bottlenecks](https://www.tensorflow.org/tensorboard/tensorboard_profiling_keras#debug_performance_bottlenecks) and check if it helps. Thanks!

@amahendrakar 
Thanks for your response. My tensorflow is 1.15.2, which running in python 3.7.8.
my code is:
`
images_array = np.asarray(images)

input_1 = self.sess.graph.get_tensor_by_name('input_1:0')

pred = self.sess.graph.get_tensor_by_name('dense_1/Softmax:0')

preds = self.sess.run(pred, feed_dict={input_1: images_array})
`

**Comment by amahendrakar** [comment]
@Dockyin,
TensorFlow 1.x is not actively supported. Could you please update TensorFlow to v2.4 and check if you are facing the same issue. Thanks!

**Comment by NeoZhangJianyu** [comment]
@Dockyin 
E5 2620 is 2012 product and with 6 cores, 32nm, support AVX.
Gold 6138T is 2017 product and with 20 cores, 14nm, support AVX512.

They have big gap in performance. So the result of your case is normal.

**Comment by Dockyin** [comment]
Solved!
The degradation is caused by virtual layer settings.!!!--_--

**Comment by NeoZhangJianyu** [comment]
@Dockyin 

It's great to find the root cause!

Thank you!
